src/Database/Bloodhound/Types.hs

```diff
@@ -604,7 +605,14 @@ data Search = Search { queryBody       :: Maybe Query
                        -- default False
                      , trackSortScores :: TrackSortScores
                      , from            :: From
-                     , size            :: Size } deriving (Eq, Show)
+                     , size            :: Size
+                     , source          :: Maybe Source } deriving (Eq, Show)
+
+data Source =
+  NoSource
+  | SourceFields FieldOrFields
```

Comment:
Well, according to the second example in the [specification](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-source-filtering.html), you can also pass in a single field as the source filter or a list of fields. I figured this is what `FieldOrFields` models so I might as well use it :) But if I misunderstood the purpose of that datatype we change it of course.